zetaclient/chains/evm/observer/inbound.go

```diff
@@ -794,6 +795,12 @@ func (ob *Observer) ObserveTSSReceiveInBlock(ctx context.Context, blockNumber ui
 				)
 			}
 		}
+		if ethcommon.HexToAddress(tx.From) == ob.TSS().EVMAddress() {
+			nonce := uint64(tx.Nonce)
+			if receipt, txx, ok := ob.checkConfirmedTx(ctx, tx.Hash, nonce); ok {
```

Comment:
We could use the similar protection logic used in `ProcessOutboundTrackers` to avoid redundant calls on `checkConfirmedTx` (which involves a group of further RPCs)

```
if !ob.IsTxConfirmed(nonce) {
    // add this tx to supplement outtx tracker
    if receipt, txx, ok := ob.checkConfirmedTx(ctx, tx.Hash, nonce); ok {
	ob.SetTxNReceipt(nonce, receipt, txx)
    }
}
```

Reply:
done https://github.com/zeta-chain/node/pull/2735/commits/e787e70208e2193009e8b9431a77672f512900ed